draft-parecki-oauth-client-id-metadata-document.md

```diff
@@ -102,12 +104,37 @@ The client metadata document URL is a JSON document containing the metadata
 of the client. The client metadata values are the values defined in
 OAuth Dynamic Client Registration ({{RFC7591}}) section 2.
 
+The client metadata document MUST contain a `client_id` property whose value
+MUST compare and match the URL of the document using simple string comparison as
+defined in [RFC3986] Section 6.2.1.
```

Comment:
This is necessary for security, as it prevents a redirect being used as a way to impersonate a client.

TBD as to the algorithm for comparison: simple string comparison or exact string matching?